patches/systemd/0002-Add-support-for-systemd-boot-for-Managarm.patch

```diff
@@ -0,0 +1,651 @@
+ #endif
+diff --git a/src/basic/stdio-util.h b/src/basic/stdio-util.h
+index 0a2239d..2e55e94 100644
+--- a/src/basic/stdio-util.h
++++ b/src/basic/stdio-util.h
+@@ -7,6 +7,7 @@
+ #include <sys/types.h>
+ 
+ #include "macro.h"
++#include "parse-printf-format.h"
+ 
+ _printf_(3, 4)
+ static inline char* snprintf_ok(char *buf, size_t len, const char *format, ...) {
+diff --git a/src/basic/time-util.c b/src/basic/time-util.c
+index 5201d5a..a4e5680 100644
+--- a/src/basic/time-util.c
++++ b/src/basic/time-util.c
+@@ -1704,7 +1704,7 @@ int time_change_fd(void) {
+ 
+         /* We only care for the cancellation event, hence we set the timeout to the latest possible value. */
+         static const struct itimerspec its = {
+-                .it_value.tv_sec = TIME_T_MAX,
++                .it_value.tv_sec = INT32_MAX, // HACK!
+         };
+ 
+         _cleanup_close_ int fd = -EBADF;
+diff --git a/src/core/exec-credential.c b/src/core/exec-credential.c
+index 6157ac4..d508df7 100644
+--- a/src/core/exec-credential.c
++++ b/src/core/exec-credential.c
+@@ -908,6 +908,7 @@ static int setup_credentials_internal(
+ 
+         if (workspace_mounted) {
+                 if (!final_mounted) {
++#ifndef __managarm__
+                         /* Make workspace read-only now, so that any bind mount we make from it defaults to
+                          * read-only too */
+                         r = mount_nofollow_verbose(LOG_DEBUG, NULL, workspace, NULL, MS_BIND|MS_REMOUNT|credentials_fs_mount_flags(/* ro= */ true), NULL);
+@@ -916,6 +917,7 @@ static int setup_credentials_internal(
+ 
+                         /* And mount it to the final place, read-only */
+                         r = mount_nofollow_verbose(LOG_DEBUG, workspace, final, NULL, MS_MOVE, NULL);
++#endif
+                 } else
+                         /* Otherwise we just get rid of the bind mount of final place */
+                         r = umount_verbose(LOG_DEBUG, workspace, MNT_DETACH|UMOUNT_NOFOLLOW);
+@@ -1036,9 +1038,11 @@ int exec_setup_credentials(
+                  * no one else sees this should be OK to do. */
+ 
+                 /* Turn off propagation from our namespace to host */
++#ifndef __managarm__
+                 r = mount_nofollow_verbose(LOG_DEBUG, NULL, "/dev", NULL, MS_SLAVE|MS_REC, NULL);
+                 if (r < 0)
+                         goto child_fail;
++#endif
+ 
+                 r = setup_credentials_internal(
+                                 context,
+diff --git a/src/core/exec-invoke.c b/src/core/exec-invoke.c
+index f72eccf..2aa6f66 100644
+--- a/src/core/exec-invoke.c
++++ b/src/core/exec-invoke.c
+@@ -226,8 +226,9 @@ static int connect_logger_as(
+         if (r < 0)
+                 return r;
+ 
+-        if (shutdown(fd, SHUT_RD) < 0)
+-                return -errno;
++        // TODO: Managarm does not implement shutdown
++        // if (shutdown(fd, SHUT_RD) < 0)
++        //         return -errno;
```

Comment:
This is potentially problematic (may break the other side of the pipe/socket/etc). We should at least log about this.

Reply:
This has been addressed.